### PR TITLE
Slack flakiness summary job

### PR DESCRIPTION
**Pull request by smnzhu**
Slack job to post flakiness summary every monday 9am

- [ ] Refactor code to be more testable and add back tests

### CODE REVIEW

File: tools/flaky_test_monitor/testdata/summary1/expected/test-result-crypto-hash-1-count-fail.json [hunk 1]
```diff
@@ -1,124 +0,0 @@
-{
```
Comment (gomisha) [inline tools/flaky_test_monitor/testdata/summary1/expected/test-result-crypto-hash-1-count-fail.json:1]: why are all the test data files deleted? They're required for all the unit tests for level 1, 2, 3

File: .github/workflows/flaky-test-monitor.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@master
+      with:
+        service_account_key: ${{ secrets.GCP_SA_KEY }}
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
```
Comment (gomisha) [inline .github/workflows/flaky-test-monitor.yml:42]: why not 1.17, like flakiness-summary.yml?

File: .github/workflows/flakiness-summary.yml [hunk 3]
```diff
@@ -1,33 +1,12 @@
...
 on:
-  push:
-    branches:
-      - master
-
-env:
-  NUM_RUNS: 1
-  GCS_BUCKET: production_ddp_flow_test_metrics
-  COMMIT_SHA: ${{ github.sha }}
+  schedule:
```
Comment (gomisha) [inline .github/workflows/flakiness-summary.yml:4]: what is this script running once a week - level 2 and 3 data upload? Why once/week?

File: .github/workflows/flaky-test-monitor.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+name: Flaky Test Monitor
+
+on:
+  schedule:
+    - cron: '*/30 * * * *' # every 30 minutes
```
Comment (gomisha) [inline .github/workflows/flaky-test-monitor.yml:5]: does it need to run every 30 minutes 24/7? What about overnight hours and weekends? running 24 times/day 24/7/365 seems excessive

File: tools/flaky_test_monitor/common/level3.go [hunk 5]
```diff
@@ -2,7 +2,7 @@ package common
 
 // models full level 3 summary of a test run from 1 (single) level 2 test run
 type TestSummary3 struct {
-	NoResults []TestResultSummary `json:"no_results"`
+	Exceptions []TestResultSummary `json:"exceptions"`
```
Comment (gomisha) [inline tools/flaky_test_monitor/common/level3.go:5]: why was this renamed?

File: tools/flaky_test_monitor/flakiness-summary.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+mkdir results
+
+for i in $(seq $NUM_DAYS); do
+    day=$(date +"%Y-%m-%d" -d "$START_DATE+$((i-1)) days")
+    gsutil -m cp -r "gs://$GCS_BUCKET/$day/*" results/
+done
+
+export FOLDER_NAME=$(echo $START_DATE | sed 's/[-]//g')-$(date +"%Y%m%d" -d "$START_DATE+$((NUM_DAYS-1)) days")
+
+upload_data () {
```
Comment (gomisha) [inline tools/flaky_test_monitor/flakiness-summary.sh:17]: some explanation / comments for the function would be helpful

File: tools/flaky_test_monitor/flakiness-summary.sh [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    mv test_outputs $1_test_outputs
+    go run level3/process_summary3_results.go level2-summary-$1.json level3
+    mv level3-summary.json level3-summary-$1.json
+    gsutil -m cp -r $1_test_outputs gs://$GCS_BUCKET/SUMMARIES/$FOLDER_NAME/
+}
+
+upload_data unit
+upload_data integration
+
+go run slack-message-generator.go level3-summary-unit.json level3-summary-integration.json
```
Comment (gomisha) [inline tools/flaky_test_monitor/flakiness-summary.sh:31]: some comments here would help - about passing data to Slackbot from level 3 summaries

File: tools/flaky_test_monitor/flakiness-summary.sh [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+curl -X POST -H 'Content-type: application/json' --data @slack-message.json $FLAKINESS_RESULTS_WEBHOOK
```
Comment (gomisha) [inline tools/flaky_test_monitor/flakiness-summary.sh:33]: some comments here would help - this is posting to the json endpoint to post to Slack?

**Comment by Kay-Zee** [comment]
Closing, as the slack integration doesn't seem like a high priority any more
